urllib3/util/ssl_.py

```diff
@@ -17,6 +18,23 @@
 
 import errno
 import warnings
+import hmac
+
+if hasattr(hmac, 'compare_digest'):
+    _const_compare_digest = hmac.compare_digest
+else:
+    def _const_compare_digest(a, b):
+        """
+        Compare two digests of equal length in constant time.
+
+        Returns True if the digests match, and False otherwise.
+        """
+        result = abs(len(a) - len(b))
+        for l, r in six.moves.zip(bytearray(a), bytearray(b)):
```

Comment:
The length check in my mind is removing a quirk -- the quirk being the different behavior on Python 2.6 and 2.7.7+. On 2.6, `_const_compare_digest('abcd', 'ab') == True`, whereas on 2.7.7+ it's False.

I'm not entirely convinced that not using six here is better, but that's your call to make -- I'll change it to plain zip.